src/ORM/DataList.php

```diff
@@ -955,10 +964,15 @@ public function byIDs($ids)
      *
      * @param int $id
      * @return DataObject
+     * @throws InvalidArgumentException
      */
     public function byID($id)
     {
-        return $this->filter('ID', $id)->first();
+        if (is_int($id)) {
+            return $this->filter('ID', (int)$id)->first();
+        } else {
+            throw new InvalidArgumentException('Incorrect param type for $id passed to byID(). Integer is expected.');
+        }
```

Comment:
Could you please refactor this so that the `return` statement comes last? You can also remove the `else` statement:

```
if (somethingwrong) {
    throw new InvalidArgumentException('...
}

return $this->filter(...
```